packages/blocks/src/_common/components/rich-text/virgo/nodes/link-node/link-popup/link-popup.ts

```diff
@@ -87,12 +87,19 @@ export class LinkPopup extends WithDisposable(LitElement) {
     assertExists(range);
 
     if (this.type !== 'view') {
-      assertExists(this.mockSelection);
-      const rangeRect = range.getBoundingClientRect();
-      this.mockSelection.style.left = `${rangeRect.left}px`;
-      this.mockSelection.style.top = `${rangeRect.top}px`;
-      this.mockSelection.style.width = `${rangeRect.width}px`;
-      this.mockSelection.style.height = `${rangeRect.height}px`;
+      const domRects = range.getClientRects();
+
+      Object.values(domRects).forEach(domRect => {
+        const mockSelection = document.createElement('div');
+        mockSelection.classList.add('mock-selection');
+        mockSelection.style.left = `${domRect.left}px`;
+        mockSelection.style.top = `${domRect.top}px`;
+        mockSelection.style.width = `${domRect.width}px`;
+        mockSelection.style.height = `${domRect.height}px`;
+
+        assertExists(this.mockSelectionContainer);
+        this.mockSelectionContainer.appendChild(mockSelection);
```

Comment:
Should the container be cleaned up before appending?

Reply:
The link-popup is a fresh element added to dom every time create / edit link called, with an empty container.
It does not update during its lifecycle. Clearing it seems not nessesary in the current implementation.